### PR TITLE
Combine multiple Service Requests into a Quotation

### PR DESCRIPTION
**Pull request by Alchez**

### CODE REVIEW

File: repairs/public/js/quotation.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+frappe.ui.form.on('Quotation', {
```
[the remark this answers is not in the document]
Comment (forellana-digithinkit) [reply repairs/public/js/quotation.js]: @Alchez Add:

/* global frappe */

At the top of your js files to quiet this warnings from eslint in codacy
You can add more globals with commas if there are more of these

File: repairs/public/js/quotation.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			frm.trigger("get_service_items");
+		}
+	},
+
+	get_service_items: (frm) => {
+		erpnext.utils.map_current_doc({
+			method: "repairs.api.make_quotation",
+			source_doctype: "Warranty Claim",
+			target: frm,
+			date_field: "complaint_date",
```
[the remark this answers is not in the document]
Comment (forellana-digithinkit) [reply repairs/public/js/quotation.js]: /*eslint camelcase: ["error", {properties: "never"}]*/

File: repairs/public/js/quotation.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	},
+
+	onload_post_render: (frm) => {
+		if (frm.doc.__islocal && frm.doc.warranty_claim) {
+			frm.trigger("get_service_items");
+		}
+	},
+
+	get_service_items: (frm) => {
+		erpnext.utils.map_current_doc({
```
[the remark this answers is not in the document]
Comment (forellana-digithinkit) [reply repairs/public/js/quotation.js]: /* globals frappe, erpnext */

File: repairs/public/js/quotation.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+frappe.ui.form.on('Quotation', {
+	refresh: (frm) => {
+		frm.add_custom_button(__('Warranty Claim'), () => {
+			frm.trigger("get_service_items");
+		}, __("Get items from"));
```
[the remark this answers is not in the document]
Comment (forellana-digithinkit) [reply repairs/public/js/quotation.js]: '/* global frappe, erpnext, __ */

**Review by DeveloperMetal — COMMENTED** [review]
See all my eslint comments to quiet codacy in regard to frappe based js code.

File: repairs/public/js/quotation.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	},
+
+	onload_post_render: (frm) => {
+		if (frm.doc.__islocal && frm.doc.warranty_claim) {
+			frm.trigger("getServiceItems");
+		}
+	},
+
+	getServiceItems: (frm) => {
```
Comment (vjFaLk) [inline repairs/public/js/quotation.js]: snake_case, cause Frappe

File: repairs/public/js/quotation.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+frappe.ui.form.on("Quotation", {
+	refresh: (frm) => {
+		frm.add_custom_button(__("Warranty Claim"), () => {
+			frm.trigger("getServiceItems");
+		}, __("Get items from"));
+	},
+
+	onload_post_render: (frm) => {
+		if (frm.doc.__islocal && frm.doc.warranty_claim) {
+			frm.trigger("getServiceItems");
```
Comment (vjFaLk) [inline repairs/public/js/quotation.js]: Explain why you're doing this.

**Pull request merged**